db/migrate/20200514214910_rename_bookmarks_controller.rb

```diff
@@ -0,0 +1,31 @@
+class RenameBookmarksController < ActiveRecord::Migration[5.2]
+  def up
+    original_controller = 'foreman_tasks_tasks'
+    original_bookmarks = Bookmark.where(controller: original_controller)
+    original_bookmarks_names = original_bookmarks.pluck(:name)
+    
+    new_controller = 'foreman_tasks/tasks'
+    new_bookmarks = Bookmark.where(controller: new_controller)
+    new_bookmarks.find_each do |new_bookmark|
+      name = new_bookmark.name
+      is_name_taken = original_bookmarks_names.include? name
+
+      if is_name_taken
+        original_bookmark = original_bookmarks.where(name: name).first
```

Comment:
To make the search more efficient, it makes sense to have the `id` of the old bookmark already in hand:
When getting original bookmarks, we can also pluck the `id` and store both in hash structure:
``` ruby
    original_bookmarks_names = Hash[original_bookmarks.pluck(:name, :id)]
```
Later, when you want to check if the name already exists, you can use the `key?` method:
``` ruby
       is_name_taken = original_bookmarks_names.key? name
```
and at last when you want to select the whole record, you can use `find` method with the `id` from hash:
``` ruby
        original_bookmark = original_bookmarks.find(original_bookmarks_names[name])
```

Reply:
that's awesome, thanks!